### ISSUE TITLE
Updater closes programs without asking

### ISSUE THREAD

**Issue by AGSPhoenix** [ShareX/ShareX#475]
I restarted my computer today and was prompted to update ShareX. Since it's usually an almost painless process with no further interaction beyond closing the ShareX window when the update completes, I accepted. The downloaded started, and I checked a few tabs I had opened in Chrome. The update downloader hit 100% and closed, and the installer popped up, and Chrome immediately closed with no warning.

When I restarted it, I was not prompted to restore my session, presumably because the installer sends its targets a close command rather than killing them. I was fortunate enough to be using an extension that tracks my session separately, and was able to restore my tabs, but others may not be so lucky.

**Comment by Jaex** [comment]
Setup file registers directshow dll which is required for screen and sound recording. But those dll used by Chrome for some reason so when setup can't register them asking for Chrome to be closed but because updater opens setup as silent then it not asks. So there is nothing I can do about it. It is kinda Chrome fault.

**Comment by Jaex** [comment]
Also it not happens every time. Example I updated to latest version when Chrome is open, still it is not closed. Don't know when Chrome interacts with directshow filters.

**Comment by WebFreak001** [comment]
Confirmed. Happens to me too

Windows 8.1 - 64 bit

**Comment by Jaex** [comment]
I have no idea how to set Inno Setup to not close Chrome. Unless someone knows how to do it, I can't fix that.

**Comment by Baegus** [comment]
I personally don't really care if it closes Chrome or not, there should just be some sort of warning and/or confirmation dialog.

**Comment by Jaex** [comment]
How am I supposed to show confirmation dialog. I didn't programmed setup software myself.

**Comment by Baegus** [comment]
Interesting, there's something about asking the user in the [Inno Setup CloseApplications documentation](http://www.jrsoftware.org/ishelp/index.php?topic=setup_closeapplications), so I'm guessing it should normally work?

**Comment by Jaex** [comment]
When not running silently it already asks.

**Comment by AGSPhoenix** [comment]
Jaex, has this been resolved? You didn't link a commit with a fix or mark the issue wontfix (aside from your initial response when I opened the ticket).

Is the issue resolved, or did you decide it wasn't worth the effort?

**Comment by Jaex** [comment]
I already marked it as "Cannot fix". What effort you talking about, I don't know how to fix it. I didn't wrote installer myself.

**Comment by AGSPhoenix** [comment]
It may be possible to include a list of updated files with each update, then check each to see if each file is in use by any application other than ShareX and warn the user before starting the installer. Not sure how easy this would be to implement though.

Alternatively, it might be possible to switch to a different installer, or write a simple installer yourself, but these would both probably be too much effort to resolve what is a relatively uncommon issue.

**Comment by Jaex** [comment]
Sorry I'm not planning to work weeks to write installer myself.

**Comment by AGSPhoenix** [comment]
And the other options?

**Comment by Jaex** [comment]
There is no option.

**Comment by AGSPhoenix** [comment]
I'm not sure I understand, I mentioned several possible solutions. Specifically, I think this might be reasonable:

"It may be possible to include a list of updated files with each update, then check each to see if each file is in use by any application other than ShareX and warn the user before starting the installer. Not sure how easy this would be to implement though."

Would this not work?

**Comment by Jaex** [comment]
I don't even know how/why Chrome using that file.

**Comment by AGSPhoenix** [comment]
It's likely that is uses it to provide its remote desktop feature; however, I don't think the reason it uses the file has any relevance.

If you don't want to implement some complicated system to detect when the installer will need to close a particular program, just add a short warning to the end of the update dialog box and leave it at that.

**Comment by Jaex** [comment]
Problem is it not everytime require to close Chrome. So if I add warning and if it don't require closing Chrome then that warning would be useless.

**Comment by AGSPhoenix** [comment]
Yeah, which is why I thought manually checking for open files might be a good idea. If you think that would be too difficult or time consuming to do, I guess we really are out of options. :/
